common/shared.c

```diff
@@ -357,7 +365,7 @@ int ft_init_ep(void)
 int ft_init_av(void)
 {
 	size_t addrlen;
-	int ret;
+	ssize_t ret;
```

Comment:
Okay, the return values are a mess.  I updated this to define 'int ret' and 'ssize_t err', and used those appropriately, well, except that ft_init_av() returns an int, but we return ssize_t from a couple of places.